apps/io-wallet-user-func/src/app/config.ts

```diff
@@ -50,6 +50,46 @@ export const ANDROID_CRL_URL =
 export const ANDROID_PLAY_INTEGRITY_URL =
   "https://www.googleapis.com/auth/playintegrity";
 
+export const MailConfig = t.type({
+  enabled: t.boolean,
+  mailSender: t.string,
+  mailhogHost: t.string,
+  mailupSecret: t.string,
+  mailupUsername: t.string,
+});
+
+export type MailConfig = t.TypeOf<typeof MailConfig>;
+
+export const getMailConfigFromEnvironment: RE.ReaderEither<
+  NodeJS.ProcessEnv,
+  Error,
+  MailConfig
+> = pipe(
+  sequenceS(RE.Apply)({
+    enabled: pipe(
+      readFromEnvironment("MailEnabled"),
+      RE.map(booleanFromString),
+      RE.orElse(() => RE.right(false)),
+    ),
+    mailSender: pipe(
+      readFromEnvironment("MailSender"),
+      RE.orElse(() => RE.right("")),
+    ),
+    mailhogHost: pipe(
+      readFromEnvironment("MailhogHost"),
+      RE.orElse(() => RE.right("")),
+    ),
+    mailupSecret: pipe(
+      readFromEnvironment("MailupSecret"),
+      RE.orElse(() => RE.right("")),
+    ),
+    mailupUsername: pipe(
+      readFromEnvironment("MailupUsername"),
+      RE.orElse(() => RE.right("")),
```

Comment:
Don't fallback to a broken default.

Reply:
If I try to do something like this:

```
readFromEnvironment("MailSender"),
RE.orElse(() => RE.right(undefined)),

```
The linter shows an error, because the readFromEnvironment() function expects an `Either<Error, string>`, not an `Either<Error, undefined>`. The solution could be to change the signature of the `readFromEnvironment()` function in the `io-wallet-common` package, but I don't understand what the side effects might be.

Reply:
You tried to remove `RE.orElse(() => RE.right(undefined))` ?

Reply:
Yes! If I remove the `RE.orElse(() => RE.right(undefined))` line, the `readFromEnvironment()` function throws an error that shows a message similar to the following:

`Error: Worker was unable to load entry point "dist/main.js": unable to find "<VARIABLE_NAME>" in node environment`

Since I only need mailhog locally to test the actual sending of the email, I can make the `MAILUP_*` variables mandatory (so they must be inserted into the JSON file configuring the function). This would also solve some other comments above. What do you think?

Reply:
You should try to make this code agnostic to your problem. The user of this code will only need to know MailUp and configure the parameters required by MailUp, they are not interested in mailhog or similar. Leaving a default value like in this case suggests that mailUp can also work without that parameter and this is sabotaged.

Reply:
Makes sense! I proceed by removing the MailhogHost environment variable, because this is an attribute not required to configure MailUp as a mail service. This solves a couple of comments, as only the username and secret are needed.